tests/e2e/metrics_test.go

```diff
@@ -126,3 +132,238 @@ func expectLearnerMetric(cx ctlCtx, procIdx int, expectMetric string) {
+			"go_memstats_mallocs_total",
+			"go_memstats_mcache_inuse_bytes",
+			"go_memstats_mcache_sys_bytes",
+			"go_memstats_mspan_inuse_bytes",
+			"go_memstats_mspan_sys_bytes",
+			"go_memstats_next_gc_bytes",
+			"go_memstats_other_sys_bytes",
+			"go_memstats_stack_inuse_bytes",
+			"go_memstats_stack_sys_bytes",
+			"go_memstats_sys_bytes",
+			"go_sched_gomaxprocs_threads",
+			"go_threads",
+			"grpc_server_handled_total",
+			"grpc_server_msg_received_total",
+			"grpc_server_msg_sent_total",
+			"grpc_server_started_total",
+			"os_fd_limit",
+			"os_fd_used",
+			"promhttp_metric_handler_requests_in_flight",
+			"promhttp_metric_handler_requests_total",
+		}
+		extraMultipleMemberClusterMetrics = []string{
+			"etcd_network_active_peers",
+			"etcd_network_peer_received_bytes_total",
+			"etcd_network_peer_sent_bytes_total",
+		}
+		extraExtensiveMetrics = []string{"grpc_server_handling_seconds"}
+	)
+
+	testCases := []struct {
+		name            string
+		options         []e2e.EPClusterOption
+		expectedMetrics []string
+	}{
+		{
+			name: "basic metrics of 1 member cluster",
+			options: []e2e.EPClusterOption{
+				e2e.WithClusterSize(1),
+			},
+			expectedMetrics: basicMetrics,
+		},
+		{
+			name: "basic metrics of 3 member cluster",
+			options: []e2e.EPClusterOption{
+				e2e.WithClusterSize(3),
+			},
+			expectedMetrics: append(basicMetrics, extraMultipleMemberClusterMetrics...),
+		},
+		{
+			name: "extensive metrics of 1 member cluster",
+			options: []e2e.EPClusterOption{
+				e2e.WithClusterSize(1),
+				e2e.WithExtensiveMetrics(),
+			},
+			expectedMetrics: append(basicMetrics, extraExtensiveMetrics...),
+		},
+		{
+			name: "extensive metrics of 3 member cluster",
+			options: []e2e.EPClusterOption{
+				e2e.WithClusterSize(3),
+				e2e.WithExtensiveMetrics(),
+			},
+			expectedMetrics: append(append(basicMetrics, extraExtensiveMetrics...), extraMultipleMemberClusterMetrics...),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			e2e.BeforeTest(t)
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+
+			epc, err := e2e.NewEtcdProcessCluster(ctx, t, tc.options...)
+			require.NoError(t, err)
+			defer epc.Close()
+
+			c := epc.Procs[0].Etcdctl()
+			for i := 0; i < 3; i++ {
+				err = c.Put(ctx, fmt.Sprintf("key_%d", i), fmt.Sprintf("value_%d", i), config.PutOptions{})
+				require.NoError(t, err)
+			}
+			_, err = c.Get(ctx, "k", config.GetOptions{})
+			require.NoError(t, err)
+
+			metricsURL, err := url.JoinPath(epc.Procs[0].Config().ClientURL, "metrics")
+			require.NoError(t, err)
+
+			mfs, err := getMetrics(metricsURL)
+			require.NoError(t, err)
+
+			var missingMetrics []string
+			for _, metrics := range tc.expectedMetrics {
+				if _, ok := mfs[metrics]; !ok {
+					missingMetrics = append(missingMetrics, metrics)
+				}
+			}
+			require.Emptyf(t, missingMetrics, "Some metrics are missing: %v", missingMetrics)
```

Comment:
Maybe just use `require.Contains` and `maps.Keys` like: `require.Contains(t, tc.expectedMetrics, slices.Collect(maps.Keys(mfs)))`

Reply:
Oh, sorry I meant `require.Subset`. Not critical but improves readability.

Reply:
Yes, `require.Subset` can improve the  readability a little bit, but it couldn't print the exactly list of missing metrics if any.